### PR TITLE
Ui fixes 2

### LINKED ISSUE #28915: Sidebar - Wielded weapons don't show it's durability
**Describe the bug**  
As title says

**To Reproduce**  
1. Wield damaged weapon

**Expected behavior**  
To see the weapon durability on the Sidebar

**Screenshots**  
image

**Versions and configuration(please complete the following information):**
Latest experimental Win64 tiles

### LINKED ISSUE #29098: SCT drawn with an offset.
**Describe the bug**  
Scrolling Combat Text (SCT) is drawn offset to the side instead of being centered on the entity it is portraying.
Happens in curses, probably also Tiles.

**To Reproduce**  
1. Spawn a monster.
2. Enable SCT.
3. Hit the monster a bunch.

**Expected behavior**  
SCT is drawn as if "coming out of" the entity being hit.

**Screenshots**  

**Versions and configuration(please complete the following information):**
 - OS: Amazon Linux 2 (4.14.88-88.73.amzn2.x86_64)
 - Game Version: 0.D-1155-0.D-1155-g40dfadfb2c (custom build after merging #28884 )
 - Graphics version: Curses
 - Mods loaded: dda + defaults

### LINKED ISSUE #29239: Self-Aware trait improperly displaying pain
**Describe the bug**  
When you have the Self-Aware trait displays pain with various ASCII characters such as quotes and parenthesis or nothing at all at low pain levels.

**To Reproduce**  
Create a character with the Self-Aware trait and recieve enough damage to cause pain.

**Expected behavior**  
Self-Aware should display pain levels as a numerical value unless that was changed previously.

**Screenshots**  
Sorry, on mobile :(

**Versions and configuration(please complete the following information):**
 - OS: Windows 8.1
 - Game Version: 0.D-1320-g341717d
 - Graphics version: Tiles with "Compact" sidebar setting
 - Mods loaded: DDA and nothing else

**Additional context**  
My assumption is it's displaying whatever ASCII character is associated with the pain number it should be displaying, but I'm not familiar enough with everything to investigate myself.

### PR DESCRIPTION
**Pull request by KorGgenT**
#### Summary

```SUMMARY: Bugfixes "fix various ui bugs and minor tweaks to sidebar option menu design"```
#### Purpose of change

Fix even more bugs.
add a hint panel to the label layout
use the input context to write the buttons you can press in the sidebar options menu instead of harcoding the text.
see commit descriptions for more detail
Fixes #29098
Fixes #28915
Fixes #29239

### CODE REVIEW

File: src/panels.cpp [hunk 1]
```diff
@@ -1932,9 +1973,12 @@ void panel_manager::draw_adm( const catacurses::window &w, size_t column, size_t
             mvwprintz( w, index, 1 + ( column_width * column ), c_yellow, ">>" );
             mvwvline( w, 1, 13, 0, 13 );
             mvwvline( w, 1, 43, 0, 13 );
-            mvwprintz( w, 1, 15, c_white, "tab: toggle panels on/off" );
-            mvwprintz( w, 2, 15, c_white, "enter: change display order" );
-            mvwprintz( w, 3, 15, c_white, "esc: exit" );
+            mvwprintz( w, 1, 15, c_light_green, trunc_ellipse( ctxt.press_x( "TOGGLE_PANEL" ), 27 ) + ":" );
+            mvwprintz( w, 2, 15, c_white, _( "toggle panels on/off" ) );
```
Comment (Night-Pryanik) [inline src/panels.cpp]:
```suggestion
            mvwprintz( w, 2, 15, c_white, _( "Toggle panels on/off" ) );
```

File: src/panels.cpp [hunk 2]
```diff
@@ -1932,9 +1973,12 @@ void panel_manager::draw_adm( const catacurses::window &w, size_t column, size_t
...
+            mvwprintz( w, 3, 15, c_light_green, trunc_ellipse( ctxt.press_x( "MOVE_PANEL" ), 27 ) + ":" );
+            mvwprintz( w, 4, 15, c_white, _( "change display order" ) );
```
Comment (Night-Pryanik) [inline src/panels.cpp]:
```suggestion
            mvwprintz( w, 4, 15, c_white, _( "Change display order" ) );
```

File: src/panels.cpp [hunk 3]
```diff
@@ -1932,9 +1973,12 @@ void panel_manager::draw_adm( const catacurses::window &w, size_t column, size_t
...
+            mvwprintz( w, 5, 15, c_light_green, trunc_ellipse( ctxt.press_x( "QUIT" ), 27 ) + ":" );
+            mvwprintz( w, 6, 15, c_white, _( "exit" ) );
```
Comment (Night-Pryanik) [inline src/panels.cpp]:
```suggestion
            mvwprintz( w, 6, 15, c_white, _( "Exit" ) );
```

File: src/panels.cpp [hunk 4]
```diff
@@ -741,26 +751,20 @@ std::pair<nc_color, std::string> pain_stat( const player &u )
     }
     // get pain string
     if( u.has_trait( trait_SELFAWARE ) && u.get_perceived_pain() > 0 ) {
-        pain_string = u.get_perceived_pain();
+        pain_string = _( "Pain " ) + to_string( u.get_perceived_pain() );
```
Comment (AMurkin) [inline src/panels.cpp]: Better to use string

File: src/panels.cpp [hunk 5]
```diff
@@ -1642,7 +1669,8 @@ void draw_weapon_classic( const player &u, const catacurses::window &w )
     werase( w );
 
     mvwprintz( w, 0, 0, c_light_gray, _( "Weapon  :" ) );
-    mvwprintz( w, 0, 10, c_light_gray, utf8_truncate( u.weapname(), getmaxx( w ) - 24 ) );
+    nc_color color = c_light_gray;
+    print_colored_text( w, 0, 10, c_light_gray, color, u.weapname( getmaxx( w ) - 24 ) );
```
Comment (AMurkin) [inline src/panels.cpp]: And everywhere else, where `color = c_light_gray`.
```suggestion
    print_colored_text( w, 0, 10, color, color, u.weapname( getmaxx( w ) - 24 ) );
```

File: src/item.cpp [hunk 6]
```diff
@@ -3030,11 +3035,16 @@ std::string item::tname( unsigned int quantity, bool with_prefix ) const
...
 
+    std::string r = ret.str();
+    if( truncate != 0 ) {
+        r = utf8_truncate( r, truncate + truncate_override );
+    }
+
     if( item_vars.find( "item_note" ) != item_vars.end() ) {
         //~ %s is an item name. This style is used to denote items with notes.
-        return string_format( _( "*%s*" ), ret.str().c_str() );
+        return string_format( _( "*%s*" ), r.c_str() );
```
Comment (AMurkin) [inline src/item.cpp]: Don't know why `_()` is here, but you may remove `.c_str()`.

**Comment by kevingranade** [comment]
SCT is still offset here.

**Comment by ifreund** [comment]
Now that the log expands to fill remaining space, wouldn't it be easier and make more sense to return to the previous behavior of ending the terrain window at the sidebar?  That would fix all currently known and yet to be discovered issues with the offset.

Pretty sure that everyone plays with the log enabled, but If not just make that space black I guess (could be done with a hidden black panel that is automatically enabled if the log is disabled)

**Comment by KorGgenT** [comment]
> Now that the log expands to fill remaining space, wouldn't it be easier and make more sense to return to the previous behavior of ending the terrain window at the sidebar? That would fix all currently known and yet to be discovered issues with the offset.
> 
> Pretty sure that everyone plays with the log enabled, but If not just make that space black I guess (could be done with a hidden black panel that is automatically enabled if the log is disabled)

i've got some pretty immediate plans, so let me try to enact them before we go back to the way it was before (already talked about it on discord, but thought i'd mention it here at least)

**Pull request merged**